### ISSUE TITLE
INTERNAL Bar subscription

### ISSUE THREAD

**Issue by rsmb7z** [nautechsystems/nautilus_trader#1021]
# Bug Report

```
2023-02-22T00:17:50.711074800Z [INF] TESTER-001.MyBaseStrategy-000: [CMD]--> Subscribe(Bar{'bar_type': BarType(EUR/USD.IDEALPRO-15-SECOND-MID-INTERNAL)}).
  File "D:\PycharmProjects\nautilus_trader\nautilus_trader\live\node.py", line 280, in run
    self.kernel.loop.run_until_complete(self.run_async())
  File "C:\Users\Javaid\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "D:\PycharmProjects\nautilus_trader\nautilus_trader\live\node.py", line 483, in run_async
    await asyncio.gather(*tasks)
  File "nautilus_trader\\live\\data_engine.pyx", line 381, in _run_cmd_queue
  File "nautilus_trader\\data\\engine.pyx", line 570, in nautilus_trader.data.engine.DataEngine._execute_command
  File "nautilus_trader\\data\\engine.pyx", line 610, in nautilus_trader.data.engine.DataEngine._handle_subscribe
  File "nautilus_trader\\data\\engine.pyx", line 860, in nautilus_trader.data.engine.DataEngine._handle_subscribe_bars
  File "nautilus_trader\\data\\client.pyx", line 568, in nautilus_trader.data.client.MarketDataClient.subscribe_bars
  File "D:\PycharmProjects\nautilus_trader\nautilus_trader\live\data_client.py", line 502, in subscribe_bars
    PyCondition.true(bar_type.is_externally_aggregated(), "aggregation_source is not EXTERNAL")
  File "nautilus_trader\\core\\correctness.pyx", line 824, in nautilus_trader.core.correctness.PyCondition.true
  File "nautilus_trader\\core\\correctness.pyx", line 57, in nautilus_trader.core.correctness.Condition.true
ValueError: aggregation_source is not EXTERNAL
```

### Expected Behavior
Engine shouldn't forward the `INTERNAL` aggregation Bar subscrption to external Data Engine, instead log in logger that Bar is already subscribed.
https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/data/engine.pyx#L854

### Actual Behavior
Thows exception if you re-subscribe `INTERNAL` bar when already subscribed.

### Steps to Reproduce the Problem

1. Subscribe the `INTERNAL` Bar when already subscribed.

**Comment by rsmb7z** [comment]
Hi @cjdsellers 
Please check at `_handle_unsubscribe_bars` as well. Same issue.

**Comment by cjdsellers** [comment]
I see the bug now, its the way that condition is setup as an `AND`. I don't think we need a warning because its reasonable/expected that subscriptions for the same bar type will keep hitting the engine.

I think the logic just needs to be tweaked to this (to avoid this downstream `ValueError` you're reporting):
```cython
        if bar_type.is_internally_aggregated()
            # Internal aggregation
            if bar_type not in self._bar_aggregators:
                self._start_bar_aggregator(client, bar_type)
        else:
            # External aggregation
            if bar_type not in client.subscribed_bars():
                client.subscribe_bars(bar_type)
```

Fix will be in my next push.

**Comment by cjdsellers** [comment]
Now fixed from 3ea96a71e6566d703b1cb105e469b797f73e6675.
